articles/143_legacy_interface_definition.md

```diff
@@ -241,3 +241,34 @@ Depending on the type the following values are valid:
 A service file contains two message definitions which are separated by a line which only contains three dashes:
 
     ---
+
+## Conversion to IDL
+Code is generated for defined interfaces to be usable by different client libraries.
+Interfaces described using the legacy format are first converted to [IDL](articles/idl_interface_definition.html).
+Code generation uses the generated file.
+
+### Mapping to IDL types
+
+| ROS type | IDL type           |
+| -------- | ------------------ |
+| bool     | boolean            |
+| byte     | octet              |
+| char     | uint8              |
```

Comment:
Do we want to add a note about IDL generators that don't support `int8` and `uint8` or are we assuming that will be correct everywhere soon?

Reply:
I wouldn't put it in the design doc since this is for the ROS-facing IDL. If a vendor doesn't support "something" we will likely take care of it by temporarily generating a "compatible" variation to feed it to their code generator. So I would call it an "implementation detail".